cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy

```diff
@@ -0,0 +1,157 @@
+/**
+    Prunes data from specified repos using method defined in repo/folder properties
+*/
+// refs
+import org.artifactory.build.promotion.PromotionConfig
+import org.artifactory.build.staging.ModuleVersion
+import org.artifactory.build.staging.VcsConfig
+import org.artifactory.exception.CancelException
+import org.artifactory.repo.RepoPathFactory
+import org.artifactory.request.Request
+import org.artifactory.util.StringInputStream
+import org.artifactory.resource.ResourceStreamHandle
+import groovy.json.JsonSlurper
+import groovy.transform.Field
+import org.artifactory.fs.ItemInfo
+import org.artifactory.repo.RepoPath
+import static com.google.common.collect.Multimaps.forMap
+// jobs to trigger based on cron or interval
+jobs {
+    // Every Min Run cron Regex  depreciation run
+    dailyDataDepreciation(cron: "0 0/1 * 1/1 * ? *") {
+        depreciateData("repositoryname-local")
+    }
+}
+def depreciateData(repoName){
+    
+    log.info "Starting data depreciation job for repository " + repoName // This will log in the LogFile
+    
+    // search given repo name for folders with property DataDeprecation = true 
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
```

Comment:
This search is looking for items in all repositories. Both `repoName` and `repoRootPath` variables are not being used. I don't believe that is the desired behavior. If you want to run the search in a specific repository add the repository name as an argument to the `itemsByProperties` method.